src/load/azext_load/data_plane/utils/utils_yaml_config.py

```diff
@@ -0,0 +1,99 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import uuid
+
+from azext_load.data_plane.utils import validators
+from azure.cli.core.azclierror import (
+    InvalidArgumentValueError,
+)
+
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def yaml_parse_autostop_criteria(data):
+    if (isinstance(data["autoStop"], str)):
+        # pylint: disable-next=protected-access
+        validators._validate_autostop_disable_configfile(data["autoStop"])
+        return {
+            "autoStopDisabled": True,
+        }
+    error_rate = data["autoStop"].get("errorPercentage")
+    time_window = data["autoStop"].get("timeWindow")
+    # pylint: disable-next=protected-access
+    validators._validate_autostop_criteria_configfile(error_rate, time_window)
+    autostop_criteria = {
+        "autoStopDisabled": False,
+    }
+    if error_rate is not None:
+        autostop_criteria["errorRate"] = error_rate
+    if time_window is not None:
+        autostop_criteria["errorRateTimeWindowInSeconds"] = time_window
+    return autostop_criteria
+
+
+def yaml_parse_splitcsv(data):
+    if not isinstance(data.get("splitAllCSVs"), bool):
+        raise InvalidArgumentValueError(
+            "Invalid value for splitAllCSVs. Allowed values are boolean true or false"
+        )
+    return data.get("splitAllCSVs")
+
+
+def validate_failure_criteria(failure_criteria):
+    parts = failure_criteria.split("(")
+    if len(parts) != 2:
+        raise ValueError(f"Invalid failure criteria: {failure_criteria}")
+    _, condition_value = parts
+    if (
+        ")" not in condition_value
+        or len(condition_value.split(")")) != 2
+        or condition_value.endswith(")")
+    ):
+        raise ValueError(f"Invalid failure criteria: {failure_criteria}")
+
```

Comment:
nit: remove extra line

Reply:
do you mean the extra new line? pylint requires 2 black lines between methods rule #E302